pkg/sources/generator/tickgen.go

```diff
@@ -323,12 +323,12 @@ func (mg *memgen) generator(ctx context.Context, rate int, timeunit time.Duratio
 	}()
 }
 
-func newreadmessage(payload []byte, offset int64) *isb.ReadMessage {
+func (mg *memgen) newreadmessage(payload []byte, offset int64) *isb.ReadMessage {
 	msg := isb.Message{
 		Header: isb.Header{
 			// TODO: insert the right time based on the generator
 			PaneInfo: isb.PaneInfo{EventTime: timefromNanos(parseTime(payload))},
-			ID:       strconv.FormatInt(offset, 10),
+			ID:       strconv.FormatInt(offset, 10) + "-" + strconv.FormatInt(int64(mg.vertexInstance.Replica), 10),
```

Comment:
use `fmt.Sprintf()`?

Reply:
generally, `fmt.Sprintf()` is slower than `strconv.XXX` family, since this is `tickgen` which is purely for load testing we could go with the faster method even though the readability is compromised?